provisioning_templates/snippet/preseed_networking_setup.erb

```diff
@@ -7,12 +7,41 @@ description: this will configure your host networking, it configures your primar
     as other configures NICs. It supports physical, VLAN and Alias interfaces. It's intended to be
     called in your preseed finish template.
 %>
+<% is_ubuntu = @host.operatingsystem.name == 'Ubuntu' -%>
+<% os_major = @host.operatingsystem.major.to_i -%>
 <% host_subnet = @host.subnet -%>
 <% host_dhcp = host_subnet.nil? ? false : host_subnet.dhcp_boot_mode? -%>
 <% host_subnet6 = @host.subnet6 -%>
 <% host_dhcp6 = host_subnet6.nil? ? false : host_subnet6.dhcp_boot_mode? -%>
 
 real=`ip -o link | awk '/<%= @host.mac -%>/ {print $2;}' | sed s/://`
+<%# Ubuntu 18.04 and newer uses netplan instead of /etc/network/interfaces -%>
+<% if is_ubuntu && os_major >= 18 -%>
+cat << EOF > /etc/netplan/01-netcfg.yaml
+# This file describes the network interfaces available on your system
+# For more information, see netplan(5).
+network:
+  version: 2
+  renderer: networkd
+  ethernets:
+    #<%= @host.primary_interface.identifier %>
+    $real:
+<% if host_dhcp6 -%>
+      dhcp6: true
+<% end -%>
+<% if host_dhcp -%>
+      dhcp4: true
+<% else -%>
+      addresses: [ <%= @host.ip %>/<%= host_subnet.cidr %><% if @host.ip6 && host_subnet6 && !host_dhcp6 -%>, <%= @host.ip6 %>/<%= host_subnet6.cidr %><% end -%> ]
```

Comment:
We have some users who use IPv6 only. I'd advise on using multiple lines YAML (for both v4 and v6) and each with their own if statements. Probably easier than to determine when to add a comma or not.